ml-agents/mlagents/utils/push_to_hf.py

```diff
@@ -0,0 +1,182 @@
+    with open(os.path.join(repo_local_path, configfile_name), "r") as yaml_in, open(
+        os.path.join(repo_local_path, "config.json"), "w") as json_out:
+        yaml_object = yaml.safe_load(yaml_in)
+        json.dump(yaml_object, json_out)
+
+
+def _generate_metadata(model_name: str, env_id: str):
+    """
+    Define the tags for the model card
+    :param model_name: name of the model
+    :param env_id: name of the environment
+    """
+    env_tag = "ML-Agents-" + env_id
+
+    metadata = {}
+    metadata["library_name"] = "ml-agents"
+    metadata["tags"] = [
+        env_id,
+        "deep-reinforcement-learning",
+        "reinforcement-learning",
+        env_tag,
+    ]
+
+    return metadata
+
+
+def _generate_model_card(repo_local_path, configfile_name, repo_id):
+    """
+    Generate the model card for the Hub
+    :param repo_local_path: local path of the repo
+    :param configfile_name: name of the yaml config file (by default configuration.yaml)
+    :param repo_id: id of the model repository from the Hugging Face Hub
+    """
+    # Step 1: Read the config.json
+    with open(os.path.join(repo_local_path, "config.json"), 'r') as f:
+        data = json.load(f)
+        # Get env_id
+        env_id = list(data["behaviors"].keys())[0]
+        # Get trainer_type
+        model_name = data["behaviors"][env_id]["trainer_type"]
+
+    # Step 2: Create the metadata
+    metadata = _generate_metadata(model_name, env_id)
+
+    # Step 3: Generate the model card
+    model_card = f"""
+  # **{model_name}** Agent playing **{env_id}**
+  This is a trained model of a **{model_name}** agent playing **{env_id}** using the [Unity ML-Agents Library](https://github.com/Unity-Technologies/ml-agents).
+  """
+
+    model_card += f"""
+  ## Usage (with ML-Agents)
+  The Documentation: https://unity-technologies.github.io/ml-agents/ML-Agents-Toolkit-Documentation/
+  We wrote a complete tutorial to learn to train your first agent using ML-Agents and publish it to the Hub:
+  - A *short tutorial* where you teach Huggy the Dog 🐶 to fetch the stick and then play with him directly in your browser: https://huggingface.co/learn/deep-rl-course/unitbonus1/introduction
+  - A *longer tutorial* to understand how works ML-Agents: https://huggingface.co/learn/deep-rl-course/unit5/introduction
+
+  ### Resume the training
+  ```
+  mlagents-learn <your_configuration_file_path.yaml> --run-id=<run_id> --resume
+  ```
+
+  ### Watch your Agent play
+  You can watch your agent **playing directly in your browser** 
+
+  1. If the environment is part of ML-Agents official environments, go to https://huggingface.co/unity
+  2. Step 1: Find your model_id: {repo_id}
+  3. Step 2: Select your *.nn /*.onnx file
+  4. Click on Watch the agent play 👀 
+  """
+  
+    return model_card, metadata
+
+
+def _save_model_card(repo_dir: Path, generated_model_card: str, metadata):
+    """Save a model card to the repository.
+    :param repo_dir: repository directory
```

Comment:
I would change this param name to `local_path`, as technically you're storing in a directory, not a repo anymore

Reply:
Changed ✅